### PR TITLE
BUGFIX: Change HTTP response to 401 for false login creds

### PR DESCRIPTION
**Pull request by tecfu**
In the event of an incorrect email or password the error response returned should be a 401.

### CODE REVIEW

File: src/controllers/users-controller.js [hunk 1]
```diff
@@ -64,7 +64,7 @@ module.exports = {
     ctx.assert(
       _.isObject(body.user) && body.user.email && body.user.password,
       422,
-      new ValidationError(["is invalid"], "", "email or password"),
+      new ValidationError(["malformed request"], "", "email or password"),
```
Comment (dimonnwc3) [inline src/controllers/users-controller.js]: where did you find the info, that message should be changed?

[here](https://github.com/gothinkster/node-express-realworld-example-app/blob/ba04b70c31af81ca7935096740a6e083563b3a4a/config/passport.js#L12) I see the same message as in this repo

File: src/controllers/users-controller.js [hunk 2]
```diff
@@ -73,15 +73,15 @@ module.exports = {
 
     ctx.assert(
       user,
-      422,
+      401,
```
Comment (dimonnwc3) [inline src/controllers/users-controller.js]: the same for conditions of missing user/invalid password

[here](https://github.com/gothinkster/node-express-realworld-example-app/blob/ba04b70c31af81ca7935096740a6e083563b3a4a/routes/api/users.js#L58) 422 is returned

I think it is logically correct. As you get this error, to let the user know, that something is wrong with the input and display a validation somehow.

**Comment by tecfu** [comment]
I urge you to review the spec: https://github.com/gothinkster/realworld/tree/master/api#errors-and-status-codes. 

It also appears that errors are not returned in the correct format.

File: src/controllers/users-controller.js (hunk 2 above)
Comment (tecfu) [reply src/controllers/users-controller.js]: Please see https://github.com/gothinkster/koa-knex-realworld-example/pull/12#issuecomment-513127294

File: src/controllers/users-controller.js (hunk 1 above)
Comment (tecfu) [reply src/controllers/users-controller.js]: Please see https://github.com/gothinkster/koa-knex-realworld-example/pull/12#issuecomment-513127294

**Pull request merged**